api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -202,8 +202,7 @@ message RateLimitQuotaBucketSettings {
     //    <envoy_v3_api_msg_service.rate_limit_quota.v3.RateLimitQuotaResponse.BucketAction.AbandonAction>`
     //    message.
     //
-    // If the field is not set, the ``ExpiredAssignmentBehavior`` time is **not limited**:
-    // it applies to the bucket until replaced by an ``active`` assignment.
```

Comment:
Is this a breaking change @tyxia?

Reply:
FWIW, note that this file is marked with a "work_in_progress = true" annotation, so I think we can probably get away with this change even if it is a breaking change.  But I'm not sure how much that's actually an issue in practice -- I'll let others answer that.

Reply:
Yes - this is intentional. Keeping it unlimited by default was an oversight. When the field is not set, we're getting into a trap of having to maintain unlimited number of buckets we track, which can be generated dynamically and be based based on user input. This opens the vector for OOM attack when the filter is not configured properly.

Reply:
Sergii, the question here is not about whether this change is important, it's about whether it's a breaking change for anyone that may already be trying to use this config.

Reply:
The other aspect @tyxia @pradeepcrao is that we need Envoy-side changes as well for the new semantics.

Reply:
@tyxia is aware that changes on the Envoy side are needed - this change was discussed with him and other folks working on the RLQS. I was the one who volunteered to do changes to the protos because @tyxia was busy with other stuff. @tyxia will be the one making Envoy changes, in a separate PR.

Reply:
There was a brief discussion quite a while ago though I didn't remember what detailed decision come out.  (We can just review this PR)

Regarding breaking change, I think it is ok since this proto is WIP and annotated with warning that[ breaking change is allowed](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rate_limit_quota_filter.html) . It is not in production yet and we are currently integrated testing with RLQS server/Control Plane.

Reply:
Thanks @tyxia. Yes, this PR should be self-describing. Please review it as is.

Reply:
OK, but we do need to ensure the stated API and Envoy implementation don't become decoupled without documenting this.